telegram/ext/_application.py

```diff
@@ -297,14 +299,16 @@ def __init__(
         self.post_stop: Optional[
             Callable[["Application[BT, CCT, UD, CD, BD, JQ]"], Coroutine[Any, Any, None]]
         ] = post_stop
-
-        if isinstance(concurrent_updates, int) and concurrent_updates < 0:
-            raise ValueError("`concurrent_updates` must be a non-negative integer!")
         if concurrent_updates is True:
             concurrent_updates = 256
-        self._concurrent_updates_sem = asyncio.BoundedSemaphore(concurrent_updates or 1)
-        self._concurrent_updates: int = concurrent_updates or 0
-
+        if isinstance(concurrent_updates, int):
+            if concurrent_updates < 0:
+                raise ValueError("`concurrent_updates` must be a non-negative integer!")
+            concurrent_updates = SimpleProcessor(concurrent_updates)
+        self._concurrent_updates_sem = asyncio.BoundedSemaphore(
+            concurrent_updates.max_concurrent_updates or 1
+        )
+        self._concurrent_updates: BaseProcessor = concurrent_updates
```

Comment:
```suggestion
        self._update_processor: BaseProcessor = concurrent_updates
```

May be a more descriptive name now :)

Actually, what do you think about `{Base, Simple}UpdateProcessor` as naming instead of just `*Processor`? 🤔

Reply:
Seems a better option to me, lets see what others think. cc @harshil21

Reply:
> more descriptive name now :)

yep good change

> what do you think about {Base, Simple}UpdateProcessor as naming instead of just *Processor?

yes would prefer that too.